File: Medium/2348_Number_of_Zero_Filled_Subarrays/Solution.java


public class Solution {
    public long zeroFilledSubarray(int[] nums) {
        long count = 0;
        long consecutiveZeros = 0;

        for (int num : nums) {
            if (num == 0) {
                consecutiveZeros++;
                count += consecutiveZeros;
            } else {
                consecutiveZeros = 0;
            }
        }

        return count;
    }

    public static void main(String[] args) {
        Solution solution = new Solution();

        // test cases 1
        int[] nums1 = {1,3,0,0,2,0,0,4};
        System.out.println(solution.zeroFilledSubarray(nums1)); // Output: 6

        // test cases 2
        int[] nums2 = {0,0,0,2,0,0};
        System.out.println(solution.zeroFilledSubarray(nums2)); // Output: 10

        // test cases 3
        int[] nums3 = {2,10,2019};
        System.out.println(solution.zeroFilledSubarray(nums3)); // Output: 0
        
    }
}

File: Medium/2348_Number_of_Zero_Filled_Subarrays/Solution.py
from typing import List, Optional


class Solution:
    def zeroFilledSubarray(self, nums: List[int]) -> int:
        count = 0
        result = 0

        for num in nums:
            if num == 0:
                count += 1
            else:
                count = 0

            result += count

        return result


if __name__ == "__main__":
    solution = Solution()

    # test cases 1
    nums1 = [1,3,0,0,2,0,0,4]
    print(solution.zeroFilledSubarray(nums1))  # Output: 6

    # test cases 2
    nums2 = [0,0,0,2,0,0]
    print(solution.zeroFilledSubarray(nums2))  # Output: 10

    # test cases 3
    nums3 = [2,10,2019]
    print(solution.zeroFilledSubarray(nums3))  # Output: 0
    